cpp/src/arrow/compute/kernels/scalar_cast_test.cc

```diff
@@ -169,7 +169,6 @@ class TestCast : public TestBase {
     ASSERT_EQ(in_values.size(), out_values.size());
     std::shared_ptr<Array> input, expected;
     if (is_valid.size() > 0) {
-      ASSERT_EQ(is_valid.size(), out_values.size());
```

Comment:
This change is out of place?

Reply:
I happen to find this check duplicate. Admittedly, this is not related to the current issue.